src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -254,7 +258,11 @@ public boolean apply(Hardware input) {
       virtualHardwareSection = updateVirtualHardwareSection(virtualHardwareSection, processorPredicate, virtualCpus + " virtual CPU(s)", BigInteger.valueOf(virtualCpus.intValue()));
       Predicate<ResourceAllocationSettingData> memoryPredicate = resourceTypeEquals(ResourceAllocationSettingData.ResourceType.MEMORY);
       virtualHardwareSection = updateVirtualHardwareSection(virtualHardwareSection, memoryPredicate, ram + " MB of memory", BigInteger.valueOf(ram.intValue()));
-
+      if (disk != null) {
+         Predicate<ResourceAllocationSettingData> diskPredicate = resourceTypeEquals(ResourceAllocationSettingData.ResourceType.DISK_DRIVE);
+         Predicate<ResourceAllocationSettingData> elementPredicate = elementNameEquals("Hard disk 1");
+         virtualHardwareSection = updateVirtualHardwareSectionDisk(virtualHardwareSection, Predicates.and(diskPredicate, elementPredicate), BigInteger.valueOf(disk.intValue()));
```

Comment:
I don't see how we'd do that easily. It's finding + modifying an attribute of an entry in HostResources, rather than modifying an element. So the code is quite different. We could have a common method that the two impls delegate to, which takes a function for modifying the `ResourceAllocationSettingData`. Shall I change it to that?